### ISSUE TITLE
RD 3.0.X: "Override Node Filter" search does nothing

### ISSUE THREAD

**Issue by gaetanquentin** [rundeck/rundeck#5034]
**Describe the bug**
When configuring job's options to run it, if i click on Override Node Filter, and click search to list nodes, it does not show any nodes.
The filter contains option variables: it must be the problem. ${option.XXX} must be empty i think

It's a problem since we cannot validate the nodes before running the job.

**My Rundeck detail**

* Rundeck version: 3.0.23
* install type: war launcher
* OS Name/version: ubuntu 18.04
* DB Type/version:postgres 10

**To Reproduce**
1. create a job with options
2. in the node filter definition, use options: environment: ${option.environment} for example
3. click to run it
4. on the configuration form: click on "Change the target nodes", and then on search

**Expected behavior**
Node list should be shown

**Screenshots**

![image](https://user-images.githubusercontent.com/5774881/60656393-a20ee180-9e4f-11e9-90dd-cca72c5e7460.png)

**Desktop (please complete the following information):**
 - OS: ubuntu 18.04
 - Browser firefox 67.04

**Comment by MegaDrive68k** [comment]
Hi @gaetanquentin 

To works as expected maybe need an enhancement that evaluates the options contained in the node filter before runs the job. The options are populated at the moment of job launch.

In fact, using options in that textbox you can see the message "The Node filters will be applied when the job is run."

Regards!

**Comment by gaetanquentin** [comment]
Yes i think it need an enhancement. Not showing target nodes is quite dangerous.

**Comment by fdevans** [comment]
Thanks for the report. As part of a cleanup of our issue tracker, we're closing issues older than 12 months, in line with Rundeck's support policy of one year from a release's date (see the [release calendar](https://docs.rundeck.com/docs/history/release-calendar.html) for currently supported versions). This isn't a judgment that the report is invalid — many of these are likely still real — but we can't verify against a version this old.

If you're still seeing this on a currently supported release (6.1.x or 5.20.x+), please open a new issue using our [bug report template](https://github.com/rundeck/rundeck/issues/new?template=bug_report.yml) with your version and reproduction steps. Please don't reopen this issue — a fresh issue keeps things tracked against a supported version and gets it in front of us properly. We're actively working on improving how issues get submitted and triaged going forward.

Thanks for your patience,
The Rundeck Team
